Mapstruct, Lombok, Maven & Eclipse

Hi,

I am using Mapstruct and Lombok in a maven based project:

    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-compiler-plugin</artifactId>
      <configuration>
        <annotationProcessorPaths>
          <path>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.16.16</version>
          </path>
          <path>
            <groupId>org.mapstruct</groupId>
            <artifactId>mapstruct-processor</artifactId>
            <version>1.2.0.Beta2</version>
          </path>
        </annotationProcessorPaths>
      </configuration>
    </plugin>

The maven compile (aka `mvn compile`) works just fine.
Within eclipse, Annotation Processing is configured using JDT APT as detailed in the mapstruct docs.

However, as soon as Eclipse is started with lomboks javaagent, annotation processing fails with an exception (see below).

Considering Lomboks & Mapstructs/APTs integration difficulties:
Is this expected/known behaviour or can it be fixed?

Many thanks.

============
```
!MESSAGE Unable to create instance of annotation processor org.mapstruct.ap.MappingProcessor
!STACK 0
**java.lang.NoClassDefFoundError: org/mapstruct/ap/spi/AstModifyingAnnotationProcessor**
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:370)
	at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
	at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
	at **org.mapstruct.ap.internal.util.AnnotationProcessorContext.findAstModifyingAnnotationProcessors(AnnotationProcessorContext.java:50)
	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.<init>(AnnotationProcessorContext.java:39)
	at org.mapstruct.ap.MappingProcessor.init(MappingProcessor.java:130)
	at** org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.discoverNextProcessor(IdeAnnotationProcessorManager.java:94)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:116)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:159)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:135)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:915)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:437)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:417)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:368)
	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.compile(IncrementalImageBuilder.java:330)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:305)
	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.build(IncrementalImageBuilder.java:135)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildDeltas(JavaBuilder.java:267)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:195)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: java.lang.ClassNotFoundException: org.mapstruct.ap.spi.AstModifyingAnnotationProcessor
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 47 more
```

@pgaschuetz Suggestions for You: install Lombok(version 1.16.10 or 1.16.12), and add "Factory Path" with Mapstruct(version 1.2.0.Beta1 or 1.2.0.Beta2). That is my workspace config, and work well.   
@filiphr Hope that will be helpful, higher version of Lombok or Mapstruct exception msg:
```
anonymousId         e4c18065-64a3-44ba-b861-f4eefe082a98
name                
email               
comment             
eclipseBuildId      4.7.0.I20170612-0950
eclipseProduct      org.eclipse.epp.package.jee.product
javaRuntimeVersion  1.8.0_91-b15
osgiWs              win32
osgiOs              Windows7
osgiOsVersion       6.1.0
osgiArch            x86_64
severity            UNKNOWN


STATUS

pluginId            org.eclipse.jdt.apt.pluggable.core
pluginVersion       1.2.0.v20170322-1054
code                1
severity            4
message             Unable to create instance of annotation processor org.mapstruct.ap.MappingProcessor
fingerprint         a8e6299ebc132c90083ff7c3f4ef8040

Exception:java.lang.NoClassDefFoundError: org/mapstruct/ap/spi/AstModifyingAnnotationProcessor
	 at java.lang.ClassLoader.defineClass1(ClassLoader.java:-2)
	 at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	 at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	 at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	 at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	 at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	 at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	 at java.security.AccessController.doPrivileged(AccessController.java:-2)
	 at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	 at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	 at java.lang.Class.forName0(Class.java:-2)
	 at java.lang.Class.forName(Class.java:348)
	 at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:370)
	 at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
	 at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
	 at HIDDEN.HIDDEN(HIDDEN:-1)
	 at HIDDEN.HIDDEN(HIDDEN:-1)
	 at HIDDEN.HIDDEN(HIDDEN:-1)
	 at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.discoverNextProcessor(IdeAnnotationProcessorManager.java:94)
	 at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:116)
	 at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:159)
	 at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:135)
	 at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:915)
	 at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:437)
	 at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:417)
	 at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:368)
	 at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:187)
	 at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:305)
	 at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:61)
	 at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:256)
	 at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:175)
	 at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	 at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	 at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	 at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	 at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:403)
	 at org.eclipse.core.internal.resources.Project$1.run(Project.java:566)
	 at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	 at org.eclipse.core.internal.resources.Project.internalBuild(Project.java:544)
	 at org.eclipse.core.internal.resources.Project.build(Project.java:112)
	 at org.eclipse.jdt.internal.ui.util.CoreUtility$BuildJob.run(CoreUtility.java:159)
	 at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: java.lang.ClassNotFoundException: org.mapstruct.ap.spi.AstModifyingAnnotationProcessor
	 at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	 at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	 at java.lang.ClassLoader.defineClass1(ClassLoader.java:-2)
	 at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	 at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	 at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	 at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	 at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	 at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	 at java.security.AccessController.doPrivileged(AccessController.java:-2)
	 at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	 at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	 at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	 at java.lang.Class.forName0(Class.java:-2)
	 at java.lang.Class.forName(Class.java:348)
	 at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:370)
	 at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
	 at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
	 at HIDDEN.HIDDEN(HIDDEN:-1)
	 at HIDDEN.HIDDEN(HIDDEN:-1)
	 at HIDDEN.HIDDEN(HIDDEN:-1)
	 at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.discoverNextProcessor(IdeAnnotationProcessorManager.java:94)
	 at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:116)
	 at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:159)
	 at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:135)
	 at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:915)
	 at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:437)
	 at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:417)
	 at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:368)
	 at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:187)
	 at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:305)
	 at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:61)
	 at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:256)
	 at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:175)
	 at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	 at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	 at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	 at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	 at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	 at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:403)
	 at org.eclipse.core.internal.resources.Project$1.run(Project.java:566)
	 at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	 at org.eclipse.core.internal.resources.Project.internalBuild(Project.java:544)
	 at org.eclipse.core.internal.resources.Project.build(Project.java:112)
	 at org.eclipse.jdt.internal.ui.util.CoreUtility$BuildJob.run(CoreUtility.java:159)
	 at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)


BUNDLES

name                org.eclipse.core.jobs
version             3.9.0.v20170322-0013

name                org.eclipse.core.resources
version             3.12.0.v20170417-1558

name                org.eclipse.core.runtime
version             3.13.0.v20170207-1030

name                org.eclipse.jdt.apt.pluggable.core
version             1.2.0.v20170322-1054

name                org.eclipse.jdt
version             3.13.0.v20170612-0950

name                org.eclipse.jdt.compiler.apt
version             1.3.0.v20170502-0408

name                org.eclipse.jdt.core
version             3.13.0.v20170516-1929

name                org.eclipse.jdt.ui
version             3.13.0.v20170511-1354
```

In my error log I get:
```
java.lang.NoClassDefFoundError: org/mapstruct/ap/spi/AstModifyingAnnotationProcessor
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:370)
	at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
	at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.findAstModifyingAnnotationProcessors(AnnotationProcessorContext.java:50)
	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.<init>(AnnotationProcessorContext.java:39)
	at org.mapstruct.ap.MappingProcessor.init(MappingProcessor.java:130)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.discoverNextProcessor(IdeAnnotationProcessorManager.java:94)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:116)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:159)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:135)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:915)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:437)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:417)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:368)
	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.compile(IncrementalImageBuilder.java:330)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:305)
	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.build(IncrementalImageBuilder.java:135)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildDeltas(JavaBuilder.java:267)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:195)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:403)
	at org.eclipse.core.internal.resources.Project$1.run(Project.java:566)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Project.internalBuild(Project.java:544)
	at org.eclipse.core.internal.resources.Project.build(Project.java:112)
	at org.eclipse.jdt.internal.ui.text.correction.SerialVersionHashOperation.computeId(SerialVersionHashOperation.java:407)
	at org.eclipse.jdt.internal.ui.text.correction.SerialVersionHashOperation.access$0(SerialVersionHashOperation.java:383)
	at org.eclipse.jdt.internal.ui.text.correction.SerialVersionHashOperation$6.run(SerialVersionHashOperation.java:366)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: java.lang.ClassNotFoundException: org.mapstruct.ap.spi.AstModifyingAnnotationProcessor
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 52 more
```

with ```1.2.0.CR1```

@filiphr this is what I get in the error log:

```
Unable to create instance of annotation processor org.mapstruct.ap.MappingProcessor

java.lang.NoClassDefFoundError: org/mapstruct/ap/spi/AstModifyingAnnotationProcessor
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:575)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:575)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1209)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1220)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1264)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1299)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1384)
	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.findAstModifyingAnnotationProcessors(AnnotationProcessorContext.java:120)
	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.<init>(AnnotationProcessorContext.java:51)
	at org.mapstruct.ap.MappingProcessor.init(MappingProcessor.java:130)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.discoverNextProcessor(IdeAnnotationProcessorManager.java:97)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:119)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:171)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:138)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotationsInternal(Compiler.java:957)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:931)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:450)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:426)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:386)
	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.compile(IncrementalImageBuilder.java:371)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:318)
	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.incrementalBuildLoop(IncrementalImageBuilder.java:186)
	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.build(IncrementalImageBuilder.java:143)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildDeltas(JavaBuilder.java:292)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:212)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:833)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:220)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:316)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:319)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:371)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:392)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.ClassNotFoundException: org.mapstruct.ap.spi.AstModifyingAnnotationProcessor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 48 more
```

session info:
```
eclipse.buildId=4.5.0.201912171052-RELEASE
java.version=11.0.6
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_CH
Framework arguments:  -product org.springframework.boot.ide.branding.sts4 -keyring /Users/domi/.eclipse_keyring
Command-line arguments:  -os macosx -ws cocoa -arch x86_64 -product org.springframework.boot.ide.branding.sts4 -keyring /Users/domi/.eclipse_keyring

This is a continuation of log file /Users/domi/work/ws/cob/.metadata/.bak_4.log
Created Time: 2020-04-12 15:16:10.847

```

@filiphr still the same:

```
java.lang.NoClassDefFoundError: org/mapstruct/ap/spi/AstModifyingAnnotationProcessor
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:575)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:575)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1209)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1220)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1264)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1299)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1384)
	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.findAstModifyingAnnotationProcessors(AnnotationProcessorContext.java:115)
	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.<init>(AnnotationProcessorContext.java:48)
	at org.mapstruct.ap.MappingProcessor.init(MappingProcessor.java:131)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.discoverNextProcessor(IdeAnnotationProcessorManager.java:97)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:119)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:171)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:138)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotationsInternal(Compiler.java:957)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:931)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:450)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:426)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:386)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:214)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:318)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:79)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:278)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:198)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:833)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:220)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:316)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:319)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:371)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:392)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.ClassNotFoundException: org.mapstruct.ap.spi.AstModifyingAnnotationProcessor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 47 more

```

@filiphr sorry, I seem to have mixed up something - no it works!!!
the stacktrace is now only part of the message and no exception is thrown anymore. Also the Mapper gets generated as expected! 👍 🥇 

Many thanks for fixing this!!!! hope to see this released soon :)

just for reference, here is the message of the log entry (severity info):
```
Failed to read AstModifyingAnnotationProcessor. Reading next processor. Reason: java.lang.NoClassDefFoundError: org/mapstruct/ap/spi/AstModifyingAnnotationProcessor  	at java.base/java.lang.ClassLoader.defineClass1(Native Method)  	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)  	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)  	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)  	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)  	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:575)  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:575)  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)  	at java.base/java.lang.Class.forName0(Native Method)  	at java.base/java.lang.Class.forName(Class.java:398)  	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1209)  	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1220)  	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1264)  	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1299)  	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1384)  	at org.mapstruct.ap.internal.util.AnnotationProcessorContext$FaultyDelegatingIterator.hasNext(AnnotationProcessorContext.java:155)  	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.findAstModifyingAnnotationProcessors(AnnotationProcessorContext.java:127)  	at org.mapstruct.ap.internal.util.AnnotationProcessorContext.<init>(AnnotationProcessorContext.java:51)  	at org.mapstruct.ap.MappingProcessor.init(MappingProcessor.java:131)  	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.discoverNextProcessor(IdeAnnotationProcessorManager.java:97)  	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:119)  	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:171)  	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:138)  	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotationsInternal(Compiler.java:957)  	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:931)  	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:450)  	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:426)  	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:386)  	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.compile(IncrementalImageBuilder.java:371)  	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:318)  	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.incrementalBuildLoop(IncrementalImageBuilder.java:186)  	at org.eclipse.jdt.internal.core.builder.IncrementalImageBuilder.build(IncrementalImageBuilder.java:143)  	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildDeltas(JavaBuilder.java:292)  	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:212)  	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:833)  	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)  	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:220)  	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)  	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:316)  	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)  	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:319)  	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:371)  	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:392)  	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)  	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)  	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)  Caused by: java.lang.ClassNotFoundException: org.mapstruct.ap.spi.AstModifyingAnnotationProcessor  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)  	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)  	... 49 more  
```

--- a/integrationtest/src/test/java/org/mapstruct/itest/tests/MavenIntegrationTest.java
+++ b/integrationtest/src/test/java/org/mapstruct/itest/tests/MavenIntegrationTest.java
@@ -156,4 +156,11 @@ public class MavenIntegrationTest {
     void usesTypeGenerationTestEclipse() {
     }
 
+    /**
+     * Tests usage of MapStruct with faulty provider of AstModifyingAnnotationProcessor.
+     */
+    @ProcessorTest(baseDir = "faultyAstModifyingAnnotationProcessorTest")
+    void faultyAstModifyingProcessor() {
+    }
+
 }

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/pom.xml
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/pom.xml
@@ -1,0 +1,50 @@
+<!--
+
+    Copyright MapStruct Authors.
+
+    Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.mapstruct.itest</groupId>
+        <artifactId>itest-faultyAstModifyingProcessor-aggregator</artifactId>
+        <version>1.0.0</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>itest-faultyAstModifyingProcessor-generator</artifactId>
+    <packaging>jar</packaging>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.mapstruct</groupId>
+            <artifactId>mapstruct-processor</artifactId>
+            <version>${mapstruct.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <compilerArgs>
+                        <compilerArg>-proc:none</compilerArg>
+                    </compilerArgs>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/AstModifyingProcessorSaysYes.java
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/AstModifyingProcessorSaysYes.java
@@ -1,0 +1,21 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.faultyAstModifyingProcessor;
+
+import javax.lang.model.type.TypeMirror;
+
+import org.mapstruct.ap.spi.AstModifyingAnnotationProcessor;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class AstModifyingProcessorSaysYes implements AstModifyingAnnotationProcessor {
+
+    @Override
+    public boolean isTypeComplete(TypeMirror type) {
+        return true;
+    }
+}

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/FaultyAstModifyingProcessor.java
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/FaultyAstModifyingProcessor.java
@@ -1,0 +1,25 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.faultyAstModifyingProcessor;
+
+import javax.lang.model.type.TypeMirror;
+
+import org.mapstruct.ap.spi.AstModifyingAnnotationProcessor;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class FaultyAstModifyingProcessor implements AstModifyingAnnotationProcessor {
+
+    public FaultyAstModifyingProcessor() {
+        throw new RuntimeException( "Failed to create processor" );
+    }
+
+    @Override
+    public boolean isTypeComplete(TypeMirror type) {
+        return true;
+    }
+}

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/FaultyStaticAstModifyingProcessor.java
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/FaultyStaticAstModifyingProcessor.java
@@ -1,0 +1,27 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.faultyAstModifyingProcessor;
+
+import javax.lang.model.type.TypeMirror;
+
+import org.mapstruct.ap.spi.AstModifyingAnnotationProcessor;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class FaultyStaticAstModifyingProcessor implements AstModifyingAnnotationProcessor {
+
+    static {
+        if ( true ) {
+            throw new RuntimeException( "Failed to initialize class" );
+        }
+    }
+
+    @Override
+    public boolean isTypeComplete(TypeMirror type) {
+        return true;
+    }
+}

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/resources/META-INF/services/org.mapstruct.ap.spi.AstModifyingAnnotationProcessor
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/generator/src/main/resources/META-INF/services/org.mapstruct.ap.spi.AstModifyingAnnotationProcessor
@@ -1,0 +1,7 @@
+# Copyright MapStruct Authors.
+#
+# Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+org.mapstruct.itest.faultyAstModifyingProcessor.UnknownAstModifyingProcessor
+org.mapstruct.itest.faultyAstModifyingProcessor.AstModifyingProcessorSaysYes
+org.mapstruct.itest.faultyAstModifyingProcessor.FaultyAstModifyingProcessor
+org.mapstruct.itest.faultyAstModifyingProcessor.FaultyStaticAstModifyingProcessor

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/pom.xml
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/pom.xml
@@ -1,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright MapStruct Authors.
+
+    Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.mapstruct</groupId>
+        <artifactId>mapstruct-it-parent</artifactId>
+        <version>1.0.0</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <groupId>org.mapstruct.itest</groupId>
+    <artifactId>itest-faultyAstModifyingProcessor-aggregator</artifactId>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>generator</module>
+        <module>usage</module>
+    </modules>
+</project>

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/pom.xml
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/pom.xml
@@ -1,0 +1,52 @@
+<!--
+
+    Copyright MapStruct Authors.
+
+    Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.mapstruct.itest</groupId>
+        <artifactId>itest-faultyAstModifyingProcessor-aggregator</artifactId>
+        <version>1.0.0</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    
+    <artifactId>itest-faultyAstModifyingProcessor-usage</artifactId>
+    <packaging>jar</packaging>
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mapstruct.itest</groupId>
+            <artifactId>itest-faultyAstModifyingProcessor-generator</artifactId>
+            <version>1.0.0</version>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <compilerArgs>
+                        <compilerArg>-XprintProcessorInfo</compilerArg>
+                        <compilerArg>-XprintRounds</compilerArg>
+                    </compilerArgs>
+                    <testCompilerArgument>-proc:none</testCompilerArgument>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/Order.java
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/Order.java
@@ -1,0 +1,22 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.faultyAstModifyingProcessor.usage;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class Order {
+
+    private String item;
+
+    public String getItem() {
+        return item;
+    }
+
+    public void setItem(String item) {
+        this.item = item;
+    }
+}

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/OrderDto.java
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/OrderDto.java
@@ -1,0 +1,22 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.faultyAstModifyingProcessor.usage;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class OrderDto {
+
+    private String item;
+
+    public String getItem() {
+        return item;
+    }
+
+    public void setItem(String item) {
+        this.item = item;
+    }
+}

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/OrderMapper.java
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/main/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/OrderMapper.java
@@ -1,0 +1,20 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.faultyAstModifyingProcessor.usage;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Filip Hrisafov
+ */
+@Mapper
+public interface OrderMapper {
+
+    OrderMapper INSTANCE = Mappers.getMapper( OrderMapper.class );
+
+    OrderDto orderToDto(Order order);
+}

--- a/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/test/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/FaultyAstModifyingTestTest.java
+++ b/integrationtest/src/test/resources/faultyAstModifyingAnnotationProcessorTest/usage/src/test/java/org/mapstruct/itest/faultyAstModifyingProcessor/usage/FaultyAstModifyingTestTest.java
@@ -1,0 +1,27 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.faultyAstModifyingProcessor.usage;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Integration test for using MapStruct with a faulty AstModifyingProcessor (i.e. when the processor cannot be loaded)
+ *
+ * @author Filip Hrisafov
+ */
+public class FaultyAstModifyingTestTest {
+
+    @Test
+    public void testMapping() {
+        Order order = new Order();
+        order.setItem( "my item" );
+
+        OrderDto dto = OrderMapper.INSTANCE.orderToDto( order );
+        assertThat( dto.getItem() ).isEqualTo( "my item" );
+    }
+}

--- a/processor/src/main/java/org/mapstruct/ap/internal/util/AnnotationProcessorContext.java
+++ b/processor/src/main/java/org/mapstruct/ap/internal/util/AnnotationProcessorContext.java
@@ -5,7 +5,10 @@
  */
 package org.mapstruct.ap.internal.util;
 
+import java.io.PrintWriter;
+import java.io.StringWriter;
 import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.ServiceLoader;
 
@@ -45,7 +48,7 @@ public class AnnotationProcessorContext implements MapStructProcessingEnvironmen
 
     public AnnotationProcessorContext(Elements elementUtils, Types typeUtils, Messager messager, boolean verbose) {
         astModifyingAnnotationProcessors = java.util.Collections.unmodifiableList(
-                findAstModifyingAnnotationProcessors() );
+            findAstModifyingAnnotationProcessors( messager ) );
         this.elementUtils = elementUtils;
         this.typeUtils = typeUtils;
         this.messager = messager;
@@ -105,18 +108,84 @@ public class AnnotationProcessorContext implements MapStructProcessingEnvironmen
         this.initialized = true;
     }
 
-    private static List<AstModifyingAnnotationProcessor> findAstModifyingAnnotationProcessors() {
+    private static List<AstModifyingAnnotationProcessor> findAstModifyingAnnotationProcessors(Messager messager) {
         List<AstModifyingAnnotationProcessor> processors = new ArrayList<>();
 
         ServiceLoader<AstModifyingAnnotationProcessor> loader = ServiceLoader.load(
                 AstModifyingAnnotationProcessor.class, AnnotationProcessorContext.class.getClassLoader()
         );
 
-        for ( AstModifyingAnnotationProcessor astModifyingAnnotationProcessor : loader ) {
-            processors.add( astModifyingAnnotationProcessor );
+        // Lombok packages an AstModifyingAnnotationProcessor as part of their jar
+        // this leads to problems within Eclipse when lombok is used as an agent
+        // Therefore we are wrapping this into an iterator that can handle exceptions by ignoring
+        // the faulty processor
+        Iterator<AstModifyingAnnotationProcessor> loaderIterator = new FaultyDelegatingIterator(
+            messager,
+            loader.iterator()
+        );
+
+        while ( loaderIterator.hasNext() ) {
+            AstModifyingAnnotationProcessor processor = loaderIterator.next();
+            if ( processor != null ) {
+                processors.add( processor );
+            }
         }
 
         return processors;
+    }
+
+    private static class FaultyDelegatingIterator implements Iterator<AstModifyingAnnotationProcessor> {
+
+        private final Messager messager;
+        private final Iterator<AstModifyingAnnotationProcessor> delegate;
+
+        private FaultyDelegatingIterator(Messager messager,
+            Iterator<AstModifyingAnnotationProcessor> delegate) {
+            this.messager = messager;
+            this.delegate = delegate;
+        }
+
+        @Override
+        public boolean hasNext() {
+            // Check the delegate maximum of 5 times
+            // before returning false
+            int failures = 5;
+            while ( failures > 0 ) {
+                try {
+                    return delegate.hasNext();
+                }
+                catch ( Throwable t ) {
+                    failures--;
+                    logFailure( t );
+                }
+            }
+
+            return false;
+        }
+
+        @Override
+        public AstModifyingAnnotationProcessor next() {
+            try {
+                return delegate.next();
+            }
+            catch ( Throwable t ) {
+                logFailure( t );
+                return null;
+            }
+        }
+
+        private void logFailure(Throwable t) {
+            StringWriter sw = new StringWriter();
+            t.printStackTrace( new PrintWriter( sw ) );
+
+            String reportableStacktrace = sw.toString().replace( System.lineSeparator(), "  " );
+
+            messager.printMessage(
+                Diagnostic.Kind.WARNING,
+                "Failed to read AstModifyingAnnotationProcessor. Reading next processor. Reason: " +
+                    reportableStacktrace
+            );
+        }
     }
 
     @Override

--- a/processor/src/test/java/org/mapstruct/ap/test/bugs/_1159/FaultyAstModifyingAnnotationProcessor.java
+++ b/processor/src/test/java/org/mapstruct/ap/test/bugs/_1159/FaultyAstModifyingAnnotationProcessor.java
@@ -1,0 +1,25 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._1159;
+
+import javax.lang.model.type.TypeMirror;
+
+import org.mapstruct.ap.spi.AstModifyingAnnotationProcessor;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class FaultyAstModifyingAnnotationProcessor implements AstModifyingAnnotationProcessor {
+
+    public FaultyAstModifyingAnnotationProcessor() {
+        throw new RuntimeException( "Faulty AstModifyingAnnotationProcessor should not be instantiated" );
+    }
+
+    @Override
+    public boolean isTypeComplete(TypeMirror type) {
+        return false;
+    }
+}

--- a/processor/src/test/java/org/mapstruct/ap/test/bugs/_1159/Issue1159Mapper.java
+++ b/processor/src/test/java/org/mapstruct/ap/test/bugs/_1159/Issue1159Mapper.java
@@ -1,0 +1,52 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._1159;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Filip Hrisafov
+ */
+@Mapper
+public interface Issue1159Mapper {
+
+    Issue1159Mapper INSTANCE = Mappers.getMapper( Issue1159Mapper.class );
+
+    CarManualDto translateManual(CarManual manual);
+
+    /**
+     * @author Filip Hrisafov
+     */
+    class CarManual {
+
+        private String content;
+
+        public String getContent() {
+            return content;
+        }
+
+        public void setContent(String content) {
+            this.content = content;
+        }
+    }
+
+    /**
+     * @author Filip Hrisafov
+     */
+    class CarManualDto {
+
+        private String content;
+
+        public String getContent() {
+            return content;
+        }
+
+        public void setContent(String content) {
+            this.content = content;
+        }
+    }
+}

--- a/processor/src/test/java/org/mapstruct/ap/test/bugs/_1159/Issue1159Test.java
+++ b/processor/src/test/java/org/mapstruct/ap/test/bugs/_1159/Issue1159Test.java
@@ -1,0 +1,60 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._1159;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.spi.AstModifyingAnnotationProcessor;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.WithServiceImplementation;
+import org.mapstruct.ap.testutil.compilation.annotation.CompilationResult;
+import org.mapstruct.ap.testutil.compilation.annotation.Diagnostic;
+import org.mapstruct.ap.testutil.compilation.annotation.ExpectedCompilationOutcome;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+import org.mapstruct.ap.testutil.runner.Compiler;
+import org.mapstruct.ap.testutil.runner.DisabledOnCompiler;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Filip Hrisafov
+ */
+@IssueKey("1159")
+@RunWith(AnnotationProcessorTestRunner.class)
+@WithClasses({
+    Issue1159Mapper.class,
+})
+@WithServiceImplementation(
+    provides = AstModifyingAnnotationProcessor.class,
+    value = FaultyAstModifyingAnnotationProcessor.class
+)
+public class Issue1159Test {
+
+    @Test
+    // The warning is not present in the Eclipse compilation for some reason
+    @DisabledOnCompiler({
+        Compiler.ECLIPSE,
+        Compiler.ECLIPSE11
+    })
+    @ExpectedCompilationOutcome(value = CompilationResult.SUCCEEDED, diagnostics = {
+        @Diagnostic(
+            kind = javax.tools.Diagnostic.Kind.WARNING,
+            messageRegExp = "Failed to read AstModifyingAnnotationProcessor. Reading next processor. Reason:.*" +
+                "Faulty AstModifyingAnnotationProcessor should not be instantiated"
+        )
+    })
+    public void shouldIgnoreFaultyAstModifyingProcessor() {
+
+        Issue1159Mapper.CarManual manual = new Issue1159Mapper.CarManual();
+        manual.setContent( "test" );
+
+        Issue1159Mapper.CarManualDto manualDto = Issue1159Mapper.INSTANCE.translateManual( manual );
+
+        assertThat( manualDto ).isNotNull();
+        assertThat( manualDto.getContent() ).isEqualTo( "test" );
+    }
+}

